editor/src/core/model/element-metadata-utils.ts

```diff
@@ -1096,6 +1102,21 @@ export const MetadataUtils = {
       intrinsicHTMLElementNamesThatSupportChildren.includes(element.name.baseVariable)
     )
   },
+  isTextEditablePropertyAccessElementAccessOrIdentifier(element: JSXElementChild): boolean {
+    // Somewhat special case these to work our way down as these 3 types tend to go together.
+    if (isJSIdentifier(element)) {
+      return true
+    } else if (isJSPropertyAccess(element)) {
+      return MetadataUtils.isTextEditablePropertyAccessElementAccessOrIdentifier(element.onValue)
+    } else if (isJSElementAccess(element)) {
+      return (
+        MetadataUtils.isTextEditablePropertyAccessElementAccessOrIdentifier(element.onValue) &&
+        MetadataUtils.isTextEditablePropertyAccessElementAccessOrIdentifier(element.element)
+      )
+    } else {
+      return false
```

Comment:
shouldn't we also return true for Arbitrary JS blocks too? othwerwise the following will not be text editable:
`<p>{element.["arbitrary" + "Js"]}</p>`

Reply:
I tried to be as conservative as possible initially, I _think_ that should be text editable however. Lets expand it in a later PR along with some tests to validate that.